client/javascript/index.js

```diff
@@ -7,634 +7,648 @@ let pickWidget = null;
-  inputFilters.className = "row";
-
-  const inputFiltersGroupSt = document.createElement("div");
-  inputFiltersGroupSt.className = "input-group col mb-3";
-
-  const inputCcyLabel = document.createElement("label");
-  inputCcyLabel.className = "input-group-text";
-  inputCcyLabel.setAttribute("for", "inputCcy");
-  inputCcyLabel.textContent = "CCY Pair";
+function createTableHeader(tr) {
+    for (let i = 0; i < tableHeadArray.length; i++) {
+        const th = document.createElement("th");
+        th.setAttribute("scope", "col");
+        th.textContent = tableHeadArray[i].name;
+        if (tableHeadArray[i].icon) {
+            const icon = document.createElement("i");
+            icon.className = "fas fa-sort";
+            th.appendChild(icon);
+        }
+        tr.appendChild(th);
+    }
+}
 
-  const inputCyy = document.createElement("select");
-  inputCyy.className = "form-select";
-  inputCyy.setAttribute("id", "inputCcy");
+function createFiltersSection(blotterButtons) {
+    const filterSubtitle = document.createElement("p");
+    filterSubtitle.className = "subtitle";
+    filterSubtitle.textContent = "FILTERS";
 
-  let tenantOptions = ["Spot", "1M", "3M"];
+    blotterButtons.appendChild(filterSubtitle);
 
-  for (let i = 0; i < tenantOptions.length; i++) {
-    const option = document.createElement("option");
-    option.setAttribute("value", tenantOptions[i]);
-    option.textContent = tenantOptions[i];
-    inputCyy.appendChild(option);
-  }
+    const verticalLine = document.createElement("div");
+    verticalLine.className = "vertical-line";
+    blotterButtons.appendChild(verticalLine);
 
-  inputFiltersGroupSt.appendChild(inputCcyLabel);
-  inputFiltersGroupSt.appendChild(inputCyy);
+    const inputFilters = document.createElement("div");
+    inputFilters.className = "row";
 
-  const inputFiltersGroupNd = document.createElement("div");
-  inputFiltersGroupNd.className = "input-group col mb-3";
+    const inputFiltersGroupSt = document.createElement("div");
+    inputFiltersGroupSt.className = "input-group col mb-3";
 
-  const inputDateLabel = document.createElement("span");
-  inputDateLabel.className = "input-group-text";
-  inputDateLabel.textContent = "Date";
+    const inputCcyLabel = document.createElement("label");
+    inputCcyLabel.className = "input-group-text";
+    inputCcyLabel.setAttribute("for", "inputCcy");
+    inputCcyLabel.textContent = "CCY Pair";
 
-  const inputDate = document.createElement("input");
-  inputDate.setAttribute("type", "date");
-  inputDate.className = "form-control";
-  inputDate.setAttribute("id", "inputDate");
-  inputDate.setAttribute("placeholder", "12/02/2018");
+    const inputCyy = document.createElement("select");
+    inputCyy.className = "form-select";
+    inputCyy.setAttribute("id", "inputCcy");
 
-  inputFiltersGroupNd.appendChild(inputDateLabel);
-  inputFiltersGroupNd.appendChild(inputDate);
+    let tenantOptions = ["Spot", "1M", "3M"];
```

Comment:
you could move all declarations at the beginning of the method, to have them all in one place